docs/reference-guide/modules/commands/pages/command-handlers.adoc

```diff
@@ -1,249 +1,926 @@
+* **Command-centric**: Define command handlers as separate methods that receive the entity as a parameter, which Axon automatically loads
+
+Both approaches achieve the same goal but reflect different design preferences.
+The entity-centric approach groups all command handling logic with the entity state, while the command-centric approach keeps command handling logic separate from state management.
+
+=== Entity-centric stateful handlers
+
+In the entity-centric model, you define a class marked with `@EventSourced` and place command handlers directly on it:
 
 [source,java]
 ----
-import org.axonframework.messaging.commandhandling.CommandHandler;
-import org.axonframework.modelling.command.AggregateIdentifier;
+import org.axonframework.extension.spring.stereotype.EventSourced;
+import org.axonframework.messaging.commandhandling.annotation.CommandHandler;
+import org.axonframework.messaging.commandhandling.annotation.Command;
+import org.axonframework.messaging.eventhandling.gateway.EventAppender;
 
-import static org.axonframework.modelling.command.AggregateLifecycle.apply;
+@Command(routingKey = "cardId") // <1>
+public record IssueCardCommand(String cardId, int amount, String shopId) {
+}
 
+@Command(routingKey = "cardId")
+public record RedeemCardCommand(String cardId, String transactionId, int amount) {
+}
+
+@EventSourced // <2>
 public class GiftCard {
 
-    @AggregateIdentifier
     private String id;
     private int remainingValue;
 
-    @CommandHandler
-    public GiftCard(IssueCardCommand cmd) {
-        apply(new CardIssuedEvent(cmd.getCardId(), cmd.getAmount()));
+    @CommandHandler // <3>
+    public static String issue(IssueCardCommand cmd, EventAppender eventAppender) {
+        if (cmd.amount() <= 0) {
+            throw new IllegalArgumentException("Amount must be positive");
+        }
+        eventAppender.append(new CardIssuedEvent(cmd.cardId(), cmd.amount()));
+        return cmd.cardId();
     }
 
-    @CommandHandler
-    public void handle(RedeemCardCommand cmd) {
-        if (cmd.getAmount() <= 0) {
+    @CommandHandler // <4>
+    public void handle(RedeemCardCommand cmd, EventAppender eventAppender) {
+        if (cmd.amount() <= 0) {
+            throw new IllegalArgumentException("amount <= 0");
+        }
+        if (cmd.amount() > remainingValue) { // <5>
+            throw new IllegalStateException("amount > remaining value");
+        }
+        eventAppender.append(new CardRedeemedEvent(id, cmd.transactionId(), cmd.amount()));
+    }
+
+    // Event sourcing handlers update state
+    @EventSourcingHandler
+    private void on(CardIssuedEvent event) {
+        this.id = event.cardId();
+        this.remainingValue = event.amount();
+    }
+
+    @EventSourcingHandler
+    private void on(CardRedeemedEvent event) {
+        this.remainingValue -= event.amount();
+    }
+
+    // Required no-arg constructor for reconstitution
+    protected GiftCard() {
```

Comment:
Pretty sure we're required to use the `@EntityCreator` annotation at all times now to mark that method that constructs the entity..

Reply:
I think this also deserves a numbered pointer in the list below to explain the necessity.